ports/stm32/timer.c

```diff
@@ -314,6 +314,35 @@ STATIC uint32_t compute_prescaler_period_from_freq(pyb_timer_obj_t *self, mp_obj
     return (prescaler - 1) & 0xffff;
 }
 
+// computes prescaler and period so TIM triggers with a period of t_num/t_den seconds
+STATIC uint32_t compute_prescaler_period_from_t(pyb_timer_obj_t *self, int32_t t_num, int32_t t_den, uint32_t *period_out) {
+    uint32_t source_freq = timer_get_source_freq(self->tim_id);
+    if (t_num <= 0 || t_den <= 0) {
+        mp_raise_ValueError("must have positive freq");
```

Comment:
Not directly relative to this PR but a quick grep shows that there's at least 10 cases in the source where a value error is raise when an argument value is negative. It might save a little code space if there was a `MP_ARG_UINT` option for the allowed arguments going in to `mp_arg_parse_all()`.